plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/price/block.tsx

```diff
@@ -96,6 +95,17 @@ export const Block = ( props: Props ): JSX.Element | null => {
 		parentName ===
 		'woocommerce/add-to-cart-with-options-grouped-product-item';
 
+	// If the block is not a descendant of the All Products block, we are
+	// already printing the styles from the PHP side (in the frontend) and the
+	// `edit.tsx` file (in the editor).
+	let styleProps = useStyleProps( props );
+	if ( ! isDescendentOfAllProductsBlock ) {
+		styleProps = {
+			className: '',
+			style: {},
+		};
+	}
```

Comment:
I've got a bit of a problem with the way this code reads. Especially to someone new to the codebase, the fact that the default `styleProps` is the one from the hook, and it is inverted inside the conditional, kinda suggests that the block is most often expected to be descendant of the All Products block, instead of _that_ being the exception, you know what I mean?

I would flip this logic, so I'd either do:


```suggestion
	const computedStyles = useStyleProps( props );
	let styleProps = {
			className: '',
			style: {},
	};
	if ( isDescendentOfAllProductsBlock ) {
		styleProps = computedStyles;
	}
```

or

```suggestion
	const computedStyles = useStyleProps( props );
	const emptyStyles = {
			className: '',
			style: {},
    };
    
    const styleProps = isDescendentOfAllProductsBlock ? computedStyles : emptyStyles;
```

I understand this is probably a nit, but I think it helps with the intent.

Reply:
Good call, I like it! I updated it in 7fd8f1e92c55c46ac481504b4f25ed1511bc48ee.